dotnet/src/Functions/Functions.OpenApi/RestApiOperationRunner.cs

```diff
@@ -221,6 +221,14 @@ internal sealed class RestApiOperationRunner
 
             throw;
         }
+        catch (OperationCanceledException ex)
+        {
+            ex.Data.Add(HttpRequestMethod, requestMessage.Method.Method);
```

Comment:
There was a discussion in one of the memory-related PRs that using strongly typed exception with dedicated properties is better than using `Exception.Data` property, see here: https://github.com/microsoft/semantic-kernel/pull/6904#discussion_r1657451436. I'm wondering if we should take it into consideration for this exception handling as well.

Reply:
Mmm, I think this is a tricky one, since adding strongly typed properties means adding new exception types.
It might be good to think through the pro's and con's of doing that. Specifically how and where developers would want to interact with failures. E.g. Do they want to do something with them in filters or where they manually invoke functions, and if so, do they even know that it's an http call that's failing and it might generate an OperationCancelledException.

Reply:
For this requirement we started with strongly typed properties and then decided to revert to using `Exception.Data` instead (in this PR https://github.com/microsoft/semantic-kernel/pull/6407).

Reply:
Created this issue to track https://github.com/microsoft/semantic-kernel/issues/7146